app/lib/actions/katello/content_view_version/import_library.rb

```diff
@@ -0,0 +1,71 @@
+# rubocop:disable Lint/SuppressedException
+module Actions
+  module Katello
+    module ContentViewVersion
+      class ImportLibrary < Actions::EntryAction
+        def plan(organization, path:, metadata:)
+          action_subject(organization)
+          unless SmartProxy.pulp_primary.pulp3_repository_type_support?(::Katello::Repository::YUM_TYPE)
+            fail ::Katello::HttpErrors::BadRequest, _("This API endpoint is only valid for Pulp 3 repositories.")
+          end
+          ::Katello::Pulp3::ContentViewVersion::Import.check!(content_view: organization.default_content_view, metadata: metadata, path: path)
+
+          version = organization.default_content_view_version
+          history = ::Katello::ContentViewHistory.create!(:content_view_version => version,
+                                                            :user => ::User.current.login,
+                                                            :status => ::Katello::ContentViewHistory::IN_PROGRESS,
+                                                            :action => ::Katello::ContentViewHistory.actions[:importing],
+                                                            :task => self.task
+                                                         )
+
+          sequence do
+            plan_action(::Actions::Pulp3::Orchestration::ContentViewVersion::Import, version, path: path, metadata: metadata)
+            concurrence do
+              version.importable_repositories.each do |repo|
+                sequence do
+                  plan_action(Actions::Pulp3::Repository::SaveVersion, repo)
+                  plan_action(Katello::Repository::MetadataGenerate, repo, :force => true)
+                  plan_action(Katello::Repository::IndexContent, id: repo.id)
+                end
+              end
+            end
+            plan_self(history_id: history.id, organization_id: organization.id)
+          end
+        end
+
+        def humanized_name
+          _("Import Default Content View")
+        end
+
+        def rescue_strategy_for_self
+          Dynflow::Action::Rescue::Skip
+        end
+
+        def finalize
+          org = ::Organization.find(input[:organization_id])
+          environment = org.library
+          view = org.default_content_view
+          version = org.default_content_view_version
+          version.update_content_counts!
+          # update errata applicability counts for all hosts in the CV & Library
+          ::Katello::Host::ContentFacet.where(:content_view_id => view.id,
+                                              :lifecycle_environment_id => environment.id).each do |facet|
+            facet.update_applicability_counts
+            facet.update_errata_status
+          end
+
+          history = ::Katello::ContentViewHistory.find(input[:history_id])
```

Comment:
Looks like we're not actually reading from `history` here, so slightly better to `update().where()` ?